extension/firefox/chrome/ShumwayCom.jsm

```diff
@@ -273,6 +291,98 @@ var ShumwayCom = {
           return;
         }
         onSystemResourceCallback = callback;
+      },
+      createLocalConnection: function(connectionName, callback) {
+        if (typeof connectionName !== 'string' ||
+            connectionName[0] !== '_' && connectionName.split(':').length !== 2) {
+          return -1; // LocalConnectionConnectResult.InvalidName
+        }
+        if (localConnectionsRegistry[connectionName]) {
+          return -2; // LocalConnectionConnectResult.AlreadyTaken
+        }
+        var connection = {
+          callback,
+          allowedSecureDomains: Object.create(null),
+          allowedInsecureDomains: Object.create(null)
+        };
+        localConnectionsRegistry[connectionName] = connection;
+        return 0; // LocalConnectionConnectResult.Success
+      },
+      hasLocalConnection: function(connectionName) {
+        return !!_getLocalConnection(connectionName);
```

Comment:
Thanks for pointing out the missing validations. I'm aware of at least some of them and intend to move all validation into dedicated functions before landing.